omp/base/index_set_kernels.cpp

```diff
@@ -148,29 +148,29 @@ GKO_INSTANTIATE_FOR_EACH_INDEX_TYPE(GKO_DECLARE_INDEX_SET_POPULATE_KERNEL);
 template <typename IndexType>
 void global_to_local(std::shared_ptr<const DefaultExecutor> exec,
                      const IndexType index_space_size,
-                     const Array<IndexType>* subset_begin,
-                     const Array<IndexType>* subset_end,
-                     const Array<IndexType>* superset_indices,
-                     const Array<IndexType>* global_indices,
-                     Array<IndexType>* local_indices, const bool is_sorted)
+                     const IndexType num_subsets, const IndexType* subset_begin,
+                     const IndexType* subset_end,
+                     const IndexType* superset_indices,
+                     const IndexType num_indices,
+                     const IndexType* global_indices, IndexType* local_indices,
+                     const bool is_sorted)
 {
 #pragma omp parallel for
-    for (size_type i = 0; i < global_indices->get_num_elems(); ++i) {
-        auto index = global_indices->get_const_data()[i];
-        GKO_ASSERT(index < index_space_size);
-        auto bucket =
-            std::distance(subset_begin->get_const_data(),
-                          std::upper_bound(subset_begin->get_const_data(),
-                                           subset_begin->get_const_data() +
-                                               subset_begin->get_num_elems(),
-                                           index));
+    for (size_type i = 0; i < num_indices; ++i) {
+        auto index = global_indices[i];
+        if (index > index_space_size) {
```

Comment:
you need to be careful with your comparison operators! This kind of issue is unlikely to be found via tests and can cause hard-to-find bugs later
```suggestion
        if (index >= index_space_size) {
```